unittest2.nim

```diff
@@ -1126,6 +1124,13 @@ template runtimeTest*(nameParam: string, body: untyped) =
       checkpoint("Unhandled exception that may cause undefined behavior: " & e.msg & " " & eTypeDesc)
       var stackTrace {.inject.} = e.getStackTrace()
       fail()
+    finally:
+      try:
+        when declared(testTeardownIMPLFlag):
+          testTeardownIMPL()
+      except CatchableError, Defect, Exception:
```

Comment:
Thanks @lchenut for this recommendation! I think now we will more protected :partying_face: